ui-tests/src/test/java/io/syndesis/qe/pages/integrations/detail/DataMapperComponent.java

```diff
@@ -0,0 +1,208 @@
+		log.info("datamapper has been loaded");
+		ElementsCollection dmColumns = this.getRootElement().findAll(Element.DM_COLLUMNS).shouldBe(size(2));
+		log.info(String.format("found %d datamapper columns", dmColumns.size()));
+		return dmColumns;
+	}
+
+	/**
+	 * Eventually returns count of found datamapper fields.
+	 *
+	 * @returns {wdpromise.Promise<boolean>}
+	 */
+	Integer fieldsCount() throws Exception {
+		ElementsCollection dmColumns = this.dataMapperColumns();
+		SelenideElement countElement = dmColumns.get(0).$(Element.LOADER_SELECTOR).shouldBe(visible);
+		String countText = countElement.getText();
+		// "77 fields" -> 77
+		String[] found = countText.split(" ");
+		if (found.length != 2) {
+			throw new Exception(String.format("failed to get files number from %s", countText));
+		}
+		return Integer.parseInt(found[0]);
+	}
+
+	void createMapping(String source, String target) throws Exception {
+		log.info(String.format("creating mapping from %s to %s", source, target));
+		ElementsCollection dmColumns = this.dataMapperColumns();
+		SelenideElement src = dmColumns.get(0);
+		SelenideElement dest = dmColumns.get(1);
+
+		this.selectMapping(source, src);
+		this.selectMapping(target, dest);
+	}
+
+	/**
+	 * Filter datamapper field element by it's name
+	 *
+	 * @param name name to find
+	 * @param fields fields array
+	 * @returns {Promise<any>} element or promise reject
+	 */
+	SelenideElement findFieldByName(String name, ElementsCollection fields) throws Exception {
+		log.info(String.format("searching field named %s", name));
+		for (SelenideElement f : fields) {
+			String fieldName = this.fieldName(f);
+			if (name.equals(fieldName)) {
+				log.info(String.format("field %s found", name));
+				return f;
+			}
+		}
+		log.warn(String.format("field %s not found between %d fields.length fields, rejecting", name, fields.size()));
+		throw new Exception(String.format("Field %s not found in given %d fields", name, fields.size()));
+	}
+
+	/**
+	 * @param mappingName for instance "User.ScreenName"
+	 * @param containerElement start searching mapping fields from here
+	 * @returns {Promise<void>}
+	 */
+	void selectMapping(String mappingName, SelenideElement containerElement) throws Exception {
+		//split and trim in one step:
+		String[] path = mappingName.trim().split("\\s*,\\s*");
+
+		ElementsCollection fields = containerElement.findAll(Element.FIELD_DETAIL).shouldBe(sizeGreaterThanOrEqual(0));
+		log.info(String.format("source has %d fields.length fields", fields.size()));
+
+		SelenideElement nextField;
+		for (String p : path) {
+			nextField = this.findFieldByName(p, fields);
+			// click on it to expand or select find correct field from list
+			log.info(String.format("Clicking on field %s", p));
+			nextField.$(Element.LABEL).shouldBe(visible).click();
+			// find all subfields for next iteration
+			fields = nextField.$$(Element.FIELD_DETAIL).shouldBe(sizeGreaterThanOrEqual(0));
```

Comment:
Isn't this shouldBe always true?

Reply:
this is here due to waiting for all elements of collection have been collected. after discussion I Updated it to:... (1)